[12.0][ADD] product_dimension_compute_volume

carmenbianca opened the pull request:
A simple module that doesn't just add an `onchange` for recomputing the volume, but makes `volume` a computed field.

Internal task: https://gestion.coopiteasy.be/web#id=8355&action=475&active_id=231&model=project.task&view_type=form&menu_id=536

carmenbianca committed:
[ADD] product_dimension_compute_volume

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca commented:
Ping @remytms for review

legalsylvain commented on product_dimension_compute_volume/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    "category": "Product",
+    "website": "https://coopiteasy.be",
+    "author": "Odoo Community Association (OCA), Coop IT Easy SCRLfs",
+    "maintainer": "Coop IT Easy SCRLfs",
```
Afaik this key is unused.

carmenbianca replied:
OK! I'll remove this.

leemannd replied:
Hello, the key is used, but it should be set if you have intentions of maintaining the module.
You set your github login and AFAIR it will allow you to merge/handle PRs related to your module. So you can work on it without beeing member of a PSC.
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_maintainer_role.rst#maintainer-role-and-responsibilities

legalsylvain commented on product_dimension_compute_volume/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    "website": "https://coopiteasy.be",
+    "author": "Odoo Community Association (OCA), Coop IT Easy SCRLfs",
+    "maintainer": "Coop IT Easy SCRLfs",
+    "maintainers": ["coopiteasy"],
```
Setting organisation doesn't work for the time being. Set a github user instead. (Carmenbianca)

carmenbianca replied:
Will fix this.

legalsylvain commented on product_dimension_compute_volume/models/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# SPDX-FileCopyrightText: 2022 Coop IT Easy SCRLfs
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
```
Copyright are usualy removed from trivial init files.

carmenbianca replied:
I am co-maintainer of [a project that advocates adding licensing blurbs to every single file](https://reuse.software) ;) I'll happily remove it here if it's unwanted, but I don't see the harm in keeping it.

rousseldenis replied:
Please at least adapt format

legalsylvain reviewed (commented):
Nitpicking review. :-)
No functional review.

carmenbianca committed:
[IMP] product_dimension_compute_volume: Add tests

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] product_dimension: Make tests fail if product_dimension_compute_volume is installed

Normally the volume is not recomputed until the onchange method is
called. This expected behaviour is now added to the tests.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca reviewed (commented):
This module should be marked as rebel module, because it interferes with the functionality of `product_dimension`. `product_dimension`'s tests should fail if this module is installed. I added a test for this in a36fe021b2b8670fdcf591424e854954b37ca65c.

I don't know how to mark a module as a rebel in Travis.

carmenbianca commented:
For some reason the tests don't fail...

rousseldenis commented on product_dimension_compute_volume/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# SPDX-FileCopyrightText: 2022 Coop IT Easy SCRLfs
```
@carmenbianca Could you adapt your headers to OCA conventions ?

https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L1

Thanks

rousseldenis commented on product_dimension_compute_volume/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+        Automatically compute the volume depending on the dimensions.""",
+    "version": "12.0.1.0.0",
+    "category": "Product",
+    "website": "https://coopiteasy.be",
```
Please adapt this

https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

rousseldenis commented on product_dimension_compute_volume/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    "depends": [
+        "product_dimension",
+    ],
+    "excludes": [],
```
Please remove void entries

rousseldenis commented on product_dimension_compute_volume/tests/test_product.py:
```diff
@@ -0,0 +1,69 @@
+        super().setUpClass()
+
+        cls.template = cls.env["product.template"].create({"name": "Template"})
+        cls.uom_m = cls.env["uom.uom"].search([("name", "=", "m")])
```
In order to avoid duplicate code, don't hesitate to create a common class

carmenbianca replied:
Thank you @rousseldenis for the code review.

I was a little lazy and copied most of this code from product_dimension. Using a common class would make sense, but it would make most sense to do it in `product_dimension`, and inherit that common class here as well. But given that you said this:

> Usually, it's better to do one PR per module, but it's ok here

Is it also OK to do that type of refactoring of tests in this PR? If not, I can just do the common class exclusively in this module.

rousseldenis replied:
> Is it also OK to do that type of refactoring of tests in this PR? If not, I can just do the common class exclusively in this module.

I think this is something we can add in ROADMAP.

carmenbianca replied:
Done :)

rousseldenis commented on product_dimension/tests/test_compute_volume.py:
```diff
@@ -11,6 +11,10 @@ def test_it_computes_volume_in_cm(self):
         self.product.product_width = 100.
         self.product.dimensional_uom_id = self.uom_cm
         self.product.volume_uom_id = self.uom_litre
+        self.assertAlmostEqual(
```
Usually, it's better to do one PR per module, but it's ok here

rousseldenis reviewed (changes requested):
@carmenbianca Thanks for this. Some changes

carmenbianca committed:
[FIX] product_dimension_compute_volume: Use OCA blurb

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[FIX] product_dimension_compute_value: __manifest__.py

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

carmenbianca committed:
[ADD] product_dimension_compute_volume: ROADMAP.rst

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

rousseldenis commented on product_dimension_compute_volume/readme/ROADMAP.rst:
```diff
@@ -0,0 +1,2 @@
+Don't Repeat Yourself in the tests. A common class in product_dimension would
```
:smiley:

remytms reviewed (approved):
LGTM

dreispt commented:
/ocabot merge nobump

carmenbianca commented:
oh wow :sweat_smile: 

edit: oops, misread this. I thought the merge failed after months of inactivity. Instead I randomly got pinged on this PR today. Hello!

remytms commented:
@dreispt Is it possible to try again a merge ? :)

dreispt commented:
/ocabot merge nobump

carmenbianca committed:
[FIX] product_dimension_compute_volume: Add setup/ files

carmenbianca commented:
@dreispt I have rebased + fixed this PR. It should work now if we try to merge again.

dreispt commented:
/ocabot merge nobump
